### PR TITLE
migrate from setup.py to pyproject.toml

### PR DESCRIPTION
**Pull request by wanghan-iapcm**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    'recommonmark',
+    'sphinx_rtd_theme>=1.0.0rc1',
+    'numpydoc',
+    'm2r2',
+    'deepmodeling-sphinx>=0.1.1',
+    'sphinx-argparse',
+]
+
+[tool.setuptools.packages.find]
+include = ["dpgen2"]
```
Comment (njzjz) [inline pyproject.toml:47]: Use `dpgen2*` to match submodules (e.g. `dpgen2.entrypoint`). Otherwise only `dpgen2` is matched.

**Review by njzjz — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (wanghan-iapcm) [reply pyproject.toml]: https://github.com/deepmodeling/dpgen2/blob/8589a93fd929ca3ca553ce300b606af7a36790c9/.github/workflows/test.yml#L22

The current behavior is testing the package without installing it. 
We may:
1. install the package.
2. restore requirement.txt, and find someway of using it in pyproject.toml, just like the what the setup.py does. 
What do you think?

Comment (njzjz) [reply pyproject.toml]: I prefer `pip install -e .` in editable mode. There is also a funny workaround in https://github.com/pypa/pip/issues/8049#issuecomment-633845028.

**Comment by wanghan-iapcm** [comment]
@njzjz could you please help fixing the issue in the doc building? thanks!

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+[project.optional-dependencies]
+docs = [
+    'sphinx',
+    'recommonmark',
+    'sphinx_rtd_theme>=1.0.0rc1',
+    'numpydoc',
+    'm2r2',
+    'deepmodeling-sphinx>=0.1.1',
+    'sphinx-argparse',
+]
```
Comment (njzjz) [inline pyproject.toml:44]: These are different from setup.py...

Comment (wanghan-iapcm) [reply pyproject.toml]: get...

**Review by njzjz — APPROVED** [review]

**Pull request merged**